radix-engine/src/updates/bottlenose.rs

```diff
@@ -1,417 +1,162 @@
-            .get_mut(CONSENSUS_MANAGER_GET_CURRENT_TIME_IDENT)
-            .unwrap();
-        export.code_hash = code_hash;
-        let function_schema = definition
-            .interface
-            .functions
-            .get_mut(CONSENSUS_MANAGER_GET_CURRENT_TIME_IDENT)
-            .unwrap();
-        function_schema.input = BlueprintPayloadDef::Static(ScopedTypeId(
-            new_schema_hash,
-            get_current_time_input_v2_type_id,
-        ));
-
-        let export = definition
-            .function_exports
-            .get_mut(CONSENSUS_MANAGER_COMPARE_CURRENT_TIME_IDENT)
-            .unwrap();
-        export.code_hash = code_hash;
-        let function_schema = definition
-            .interface
-            .functions
-            .get_mut(CONSENSUS_MANAGER_COMPARE_CURRENT_TIME_IDENT)
-            .unwrap();
-        function_schema.input = BlueprintPayloadDef::Static(ScopedTypeId(
-            new_schema_hash,
-            compare_current_time_input_v2_type_id,
-        ));
-
-        scrypto_encode(
-            &PackageBlueprintVersionDefinitionVersions::V1(definition)
-                .into_versioned()
-                .into_locked_substate(),
-        )
-        .unwrap()
-    };
+    /// Introduces the account locker blueprint.
+    pub add_locker_package: UpdateSetting<()>,
 
-    let bp_definition_partition_num = reader
-        .get_partition_of_collection(
-            &consensus_mgr_pkg_node_id,
-            ObjectModuleId::Main,
-            PackageCollection::BlueprintVersionDefinitionKeyValue.collection_index(),
-        )
-        .unwrap();
+    /// Makes some behavioral changes to the try_deposit_or_refund (and batch variants too) method
+    /// on the account blueprint.
+    pub fix_account_try_deposit_or_refund_behaviour: UpdateSetting<()>,
 
-    let code_vm_type_partition_num = reader
-        .get_partition_of_collection(
-            &consensus_mgr_pkg_node_id,
-            ObjectModuleId::Main,
-            PackageCollection::CodeVmTypeKeyValue.collection_index(),
-        )
-        .unwrap();
+    /// Moves various protocol parameters to state.
+    pub move_protocol_params_to_state: UpdateSetting<ProtocolParamsSettings>,
 
-    let code_partition_num = reader
-        .get_partition_of_collection(
-            &consensus_mgr_pkg_node_id,
-            ObjectModuleId::Main,
-            PackageCollection::CodeOriginalCodeKeyValue.collection_index(),
-        )
-        .unwrap();
+    /// Adds an XRD vault to the access controller for locking fees.
+    pub update_access_controller_to_add_xrd_fee_vault: UpdateSetting<()>,
+}
 
-    let schema_partition_num = reader
-        .get_partition_of_collection(
-            &consensus_mgr_pkg_node_id,
-            ObjectModuleId::Main,
-            PackageCollection::SchemaKeyValue.collection_index(),
-        )
-        .unwrap();
+#[derive(Clone)]
+pub struct ProtocolParamsSettings {
+    pub network_definition: NetworkDefinition,
```

Comment:
Looks like this is the only use case of the `T` associated with `UpdateSetting`.

Reply:
Yes but as explained in the other comment, if genesis is anything to go by; I think as we redefine various constants we’ll want to be able to reconfigure certain updates (e.g. consensus related config) quite flexibly on testnets and in integration tests.

Reply:
I can see the merits of David's approach to this where the protocol is easily configurable through these settings.

In my opinion, these settings can have a right and wrong usage so I think we will need to be careful about using them. Here are some examples:

* Good: Using them to configure everything about genesis, using them to set values that need to be different on different environments. E.g., the amount of time that it takes to unstake needs to be different between mainnet and test networks.
* Bad: Using them to change what a protocol update does. As an example, let's say that we define that one of the protocol updates in anemone is an update to the validator creation fee to become 100 XRD instead of 1000 XRD. Having a setting that allows for tweaking the XRD amount on that on the protocol update feels wrong to me personally since the protocol update is now no longer concrete or strongly defined. If a setting for that were to be added then this protocol update's definition would change from "Update to the validator creation fee to become 100 XRD instead of 1000 XRD" to a generic "Update to the validator creation fee" since its something set by a setting.

But, as I said, I can see the merit of this approach and the need we have for it especially different networks requiring somewhat of different configuration. My recommendation would be that we move forward with this approach and we examine its uses in the future. I think the first occurrence of that might be when we move genesis to become a protocol update and the various configuration that it would have.